gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspExtension.kt

```diff
@@ -37,6 +48,16 @@ abstract class KspExtension @Inject constructor(project: Project) {
 
     open val arguments: Map<String, String> get() = apOptions.get()
 
+    init {
+        @Suppress("LeakingThis")
+        useKsp2.convention(
+            project.providers
```

Comment:
You can inject a providerfactory and Objectfactory so you can initialize the whole property inline

Reply:
I typically prefer to use abstract properties. You do have to to suppress the lint error but I like it better anyway

Reply:
Then I think it's at least better to move the convention setting to the task registration block? This code also accesses `Task.getProject()`, which is a generally soft-deprecated API even if you're using it in the one safe place here.

Reply:
This is an extension, not a task. The convention setting could be moved to the plugin, I have no strong opinion

Reply:
Moved the convention to the plugin